src/webapp/components/theme-list/ThemePermissionsDialog.tsx

```diff
@@ -16,49 +16,69 @@ export interface ThemePermissionsDialogProps {
 export function ThemePermissionsDialog(props: ThemePermissionsDialogProps) {
     const { themeId, onClose, rows: themes, onChange } = props;
     const { compositionRoot } = useAppContext();
+    const [errors, setErrors] = useState<string[]>([]);
 
     const search = useCallback((query: string) => compositionRoot.users.search(query), [compositionRoot]);
 
     const buildMetaObject = useCallback(
         (id: Id) => {
-            const buildSharings = (type: PermissionType) => {
+            const buildSharings = () => {
                 const theme = themes.find(theme => theme.id === id);
-                return type === "user" ? theme?.sharing?.users : theme?.sharing?.userGroups;
+                return {
+                    users: theme?.sharing?.users,
+                    userGroups: theme?.sharing?.userGroups,
+                    external: theme?.sharing?.external,
+                    public: theme?.sharing?.public,
+                };
             };
 
             return {
                 meta: {
-                    allowPublicAccess: false,
+                    allowPublicAccess: true,
                     allowExternalAccess: false,
                 },
                 object: {
                     id: "",
                     displayName: i18n.t("Access to Themes"),
-                    externalAccess: false,
-                    publicAccess: "",
-                    userAccesses: buildSharings("user"),
-                    userGroupAccesses: buildSharings("userGroup"),
+                    externalAccess: buildSharings().external,
+                    publicAccess: buildSharings().public,
+                    userAccesses: buildSharings().users,
+                    userGroupAccesses: buildSharings().userGroups,
```

Comment:
It's a bit weird to call the same function repeatedly when the arguments do not change. I'd say we need a `const sharing = ...` above and just use it.